simulator/consensus_sim_server.go

```diff
@@ -197,23 +196,23 @@ func main() {
 	if *cpuprofile != "" {
 		f, err := os.Create(*cpuprofile)
 		if err != nil {
-			log.Fatal(err)
+			logger.Fatal().Err(err).Msgf("failed to create file: %v", err)
```

Comment:
Since you already add error to the error field. You don't need to add it to msg again. Same for the following.

Reply:
ok